From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Sat, 8 Oct 2022 10:40:12 +0200
Subject: [PATCH] HACK: Allwinner: Don't set up PMIC

There are two issues:
1. Regulators must be bring up in certain order with delays between
   steps which are not considered here.
2. PMIC setup fails on board without PMIC, like Tanix TX6.

Since LibreELEC doesn't need things that need additional regulators to
be set up in U-Boot like networking, disable PMIC completely in order to
avoid above issues.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 plat/allwinner/common/sunxi_bl31_setup.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/plat/allwinner/common/sunxi_bl31_setup.c b/plat/allwinner/common/sunxi_bl31_setup.c
index a32124a182a8..34a68dd80e70 100644
--- a/plat/allwinner/common/sunxi_bl31_setup.c
+++ b/plat/allwinner/common/sunxi_bl31_setup.c
@@ -177,8 +177,6 @@ void bl31_platform_setup(void)
 	if (soc_id == SUNXI_SOC_A64 || soc_id == SUNXI_SOC_H5)
 		mmio_write_32(SUNXI_CCU_BASE + 0x5c, 0x1);
 
-	sunxi_pmic_setup(soc_id, fdt);
-
 	INFO("BL31: Platform setup done\n");
 }
 
